### DISCUSSION TITLE
VEML 6070 UV, how to use it?

### DISCUSSION THREAD

**Discussion by Airwolf25400** [arendst/Tasmota#11458] [Q&A] [answered]
Hi everybody,
I want to measure the UV index with a VEML6070 module. I have configured my D1 / D2 ports in SCL / SDA, I am in generic module (18). I cannot see the UV Index value in the TASMOTA main menu. I do not understand why. Has anyone already configured TASMOTA for this type of sensor?
I am on MAC
Thank you

<img width="420" alt="Capture d’écran 2021-03-22 à 22 14 34" src="https://user-images.githubusercontent.com/75117096/112329769-c364d080-8cb7-11eb-98ca-67021093e4e3.png">

<img width="431" alt="Capture d’écran 2021-03-24 à 15 45 27" src="https://user-images.githubusercontent.com/75117096/112330099-09ba2f80-8cb8-11eb-802a-d85090a591d7.png">

**Comment by Airwolf25400** [comment, marked as answer]
I attach you the assembly of my project

![IMG_8018](https://user-images.githubusercontent.com/75117096/112330670-86e5a480-8cb8-11eb-9a40-544569bc9cee.jpg)

**Reply by barbudor** [reply]
Which exact [binary](https://github.com/arendst/Tasmota/blob/development/BUILDS.md) have you flashed ? => needs Tasmota-sensor.bin(.gz)
Have you enabled the driver with [I2CDriver](https://tasmota.github.io/docs/Commands/#i2cdriver) ?
You should also disable all unused driver with the same command to avoid I2C addresses conflicts between drivers dedicated for devices that are using the same address

**Reply by Airwolf25400** [reply]
I flashed with the tasmota-FR.bin

**Reply by sfromis** [reply]
If you need both the sensor and French as language, you'll need to [compile your own binary](https://tasmota.github.io/docs/Compile-your-build/) to combine these settings.

**Reply by Airwolf25400** [reply]
I just flashed with the tasmota-sensor binary and it works great. Thanks for the advice.
On the other hand, can I put TASMOTA in French?

**Reply by sfromis** [reply]
Yes, if you compile your own binary, as I just mentioned.

**Reply by Airwolf25400** [reply]
> If you need both the sensor and French as language, you'll need to [compile your own binary](https://tasmota.github.io/docs/Compile-your-build/) to combine these settings.

I don't know how can compile my own binary on Mac OS

**Reply by sfromis** [reply]
Read the link I posted. It has suggestions for compiling online via your browser. You need not install anything on your own computer.

**Reply by ascillato** [reply]
Click on the link and you will learn how to compile. It is easy and fast.

**Reply by Airwolf25400** [reply]
thanks for your answers I'll try to compile my binary

**Comment by Airwolf25400** [comment]
Hello
I managed to compile my binary thank you for your information. Do you have a way to save the firmware once configured in my ESP8266.

**Reply by barbudor** [reply]
Not sure to understand
Are you asking how to upload the binary to the ESP ? You can just do Firmware Upgrade form the menu and use the tasmota.bin.gz that you just compiled.
If you have already done that and want to backup the configuration, you can also to that from the web menu Configuration => Backup Configuration which will provide you with a binary file of the settings
You may also want to look at the tool decode-config (github:tasmota/decode-config) which allows to save the configuration as a JSON file
